ckan/logic/action/update.py

```diff
@@ -294,13 +294,12 @@ def package_update(context, data_dict):
     else:
         rev.message = _(u'REST API: Update object %s') % data.get("name")
 
-    #avoid revisioning by updating directly
-    model.Session.query(model.Package).filter_by(id=pkg.id).update(
-        {"metadata_modified": datetime.datetime.utcnow()})
-    model.Session.refresh(pkg)
-
     pkg = model_save.package_dict_save(data, context)
 
+    # Update metadata modified ONLY when pkg metadata was changed.
+    if model.Session.is_modified(pkg):
```

Comment:
we need to update metadata_modified when any of the extras or resources in this package have been modified as well